src/main/java/htsjdk/variant/variantcontext/writer/VCFWriter.java

```diff
@@ -195,6 +201,32 @@ public static VCFHeader writeHeader(VCFHeader header,
     }
 
     /**
+     * Given a header and a target output version, see if the header's version is compatible with the
+     * requested version.
+     * @param header
+     * @param versionLine
+     */
+    private static void validateHeaderVersion(final VCFHeader header, final String versionLine) {
+        Utils.nonNull(header);
+        Utils.nonNull(versionLine);
+
+        final VCFHeaderVersion vcfVersion = header.getHeaderVersion();
+        if (!vcfVersion.equals(VCFHeaderVersion.getHeaderVersion(versionLine))) {
+            final String message = String.format("Attempt to write a version %s VCF header to a version %s VCF output",
+                    vcfVersion.getVersionString(),
+                    versionLine);
+            if (VCFHeaderVersion.versionsAreCompatible(VCFHeaderVersion.getHeaderVersion(versionLine), vcfVersion)) {
+                if (VCFUtils.getStrictVCFVersionValidation()) {
+                    throw new TribbleException(message);
+                }
+            }
+            if (VCFUtils.getVerboseVCFLogging()) {
+                logger.warn(message);
```

Comment:
Not quite sure I get what you're saying. Are you saying to only log this message if the versions are incompatible ?